aws/services/cognito/resource.ftl

```diff
@@ -107,15 +107,58 @@
     ]
 [/#function]
 
-[#function getUserPoolSchemaObject name, datatype, mutable, required]
+[#function getUserPoolSchemaObject name datatype mutable required constraints ]
+
+    [#local schema =
+        {
+            "Name" : name,
+            "AttributeDataType" : datatype,
+            "Mutable" : mutable,
+            "Required" : required
+        }]
+
+    [#switch datatype?lower_case ]
```

Comment:
Does this require a default case or other validation of the datatype?

Reply:
What do you mean by default case? the datatype variable will be forced to lowercase here. 
This function can only be called from a setup routine which would handle the validation of whats provided by the user

Reply:
> the datatype variable will be forced to lowercase here.
I mean as in

```
[#default]
    [@fatal
       message="Unexpected data type"
       context=datatype
    /]
    [#break]

```

Reply:
Updated with default